src/effective-dart/usage.md

```diff
@@ -350,13 +350,18 @@ then it probably does make sense to have a separate boolean field.
 
 Checking that a nullable variable is not equal to `null` promotes the variable
 to a non-nullable type. That lets you access members on the variable and pass it
-to functions expecting a non-nullable type. Unfortunately, promotion is only
-sound for local variables and parameters, so fields and top-level variables
-aren't promoted.
+to functions expecting a non-nullable type.
 
-One pattern to work around this is to assign the field's value to a local
-variable. Null checks on that variable do promote, so you can safely treat
-it as non-nullable.
+Type promotion is only sound, however, for local variables, parameters, and
+private final fields. Instances that are open to manipulation, whether
```

Comment:
I found this a bit confusing because "instances" has a technical meaning that's different from what you're trying to communicate here. E.g., an object on the heap is often said to be an "instance" of a certain class. But we're not talking about heap objects being manipulated here, we're talking about things like non-final fields, which might _refer_ to different heap objects from time to time, and getters, which might return a different heap reference from time to time.

Maybe change "instances" to "values"? It's not really any more precise than "instances", but at least it avoids the confusing connotation.